accepted/0000-selective-versions-resolutions.md

```diff
@@ -0,0 +1,303 @@
+latest available version for typescript is `2.3.2` then a warning is issued.
+
+The rational behind that is that since the `yarn.lock` file is only modified
+by the user (via yarn commands), then a warning will always be issued before
+such a situation happens and is written to the `yarn.lock` file.
+
+## Warnings in logs
+
+yarn should warn about the following situations:
+- Unused resolutions
+- Incompatible resolutions: see the above section about `yarn.lock`.
+Incompatible resolutions should be accepted but warned about since it could
+lead to unwanted behaviour.
+- Broadening specifications: see above about non-exact specifications. This
+actually falls under the umbrella of incompatible resolutions.
+- ? (see open questions below)
+
+# How We Teach This
+
+This won't have much impact as it extends the current behaviour by adding
+functionality.
+
+The only breaking change is that `resolutions` is being considered all the time,
+but that won't surprise people, this will make yarn behaviour simply more
+consistent than before (see the comment on `--flat` above).
+
+The term "resolution" has the same meaning as before, but it is not under the
+sole control of yarn itself anymore, but also under the control of the user
+now.
+
+This is an advanced use of yarn, so new users don't really have to know about
+it in the beginning.
+
+# Drawbacks
+
+## Teaching
+
+It makes yarn behaviour a bit more complex, even though more useful. So it
+can be difficult for users to wrap their head around it. The RFC submitter has
+seen it happen many times with maven, which is quite complex but complete in
+its dependency management. Users would get confused and it can take time to
+understand the implications of manipulation the `resolutions` field (even
+though, the chosen solution, compared to the alternatives below, is much
+simpler).
+
+## Package management paradigm
+
+Yarn and npm users are highly used to the idea that a dependency can be
+present many times in the `node_module`, depending on which package needs it.
+This has advantages and disadvantages, but it is one of the specificity of the
+npm ecosystem package management.
+
+In this light, taking such as design decision puts yarn a bit farther to such
+way of doing thing, and it could be considered a bad direction to go toward.
+
+Some of the alternatives below actually take this into consideration, but are
+a bit more complex in terms of expressiveness, so were not chosen by the RFC
+submitter (see open questions below too).
+
+# Alternatives
+
+There is at least one alternative to the proposed solution, more complex but
+more expressive.
+
+## Nested dependencies resolution per dependency
+
+Starting from an example, this solution would take the following form in the
+`package.json` file:
+```json
+  "devDependencies": {
+    "@angular/cli": "1.0.3",
+    "typescript": "2.3.2"
+  },
+  "resolutions": {
+    "@angular/cli": {

```

Comment:
I would prefer nesting to be an option.
JS project, unlike Java, have hundreds of dependencies, it is not rare to have several versions of the same library at different levels and this is fine.
Sometimes I'd need a fine control to override resolution only for a specific dependency path